source/common/router/router.h

```diff
@@ -147,26 +148,21 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   }
 
   /**
-   * Set a pseudo-random cookie with the name @param key and TTL @param max_age
-   * to be sent to the downstream host.
+   * Set a computed cookie to be sent with the downstream headers.
+   * @param key supplies the size of the cookie
+   * @param max_age the lifetime of the cookie
    * @return std::string the value of the new cookie
-   *
-   * marked const so it can be called from hashKey(), and because all mutated
-   * state is only visible when calling onUpstreamHeaders()
    */
-  std::string addDownstreamSetCookie(const std::string& key, std::chrono::seconds max_age) const {
+  std::string addDownstreamSetCookie(const std::string& key, std::chrono::seconds max_age) {
     // The cookie value should be the same per connection so that if multiple
     // streams race on the same path, they all receive the same cookie.
     // Since the downstream port is part of the hashed value, multiple HTTP1
     // connections can receive different cookies if they race on requests.
     std::string value;
     const Network::Connection* conn = downstreamConnection();
-    if (conn) {
-      value = conn->remoteAddress().asString() + conn->localAddress().asString();
-    } else {
-      // TODO(akonradi) talk to mattklein123 and figure out when this can happen
-      ENVOY_LOG(warn, "Downstream connection was null while trying to make a routing cookie");
-    }
+    // need to check for null conn if this is ever used by Http::AsyncClient in the fiture
+    ASSERT(conn);
```

Comment:
nit: You can remove this ASSERT() as it will crash in an obvious way on the next line.